server/stream.go

```diff
@@ -1780,6 +1780,27 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 	// Now update config and store's version of our config.
 	mset.cfg = *cfg
 
+	// Check for changes to the retention policy. If we are moving to interest-based
+	// retention then we will need to check the ack floor of all consumers.
+	if ocfg.Retention != cfg.Retention && cfg.Retention == InterestPolicy {
+		toUpdate := make([]*consumer, 0, len(mset.consumers))
+		for _, c := range mset.consumers {
```

Comment:
One important piece, if the consumer is retention interest, then its replication factor must be same as stream, so need to adjust that here and add in an R1 consumer to the test above etc.